server/filters.go

```diff
@@ -0,0 +1,154 @@
+package server
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/Masterminds/semver"
+	pbVersion "github.com/Percona-Lab/percona-version-service/versionpb"
+	"github.com/diegoholiveira/jsonlogic"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	recommended = "recommended"
+	latest      = "latest"
+)
+
+func pxcFilter(versions map[string]*pbVersion.Version, apply string, current string) error {
+	if len(versions) == 0 {
+		return status.Error(codes.Internal, "no versions to filter")
+	}
+
+	keys := make([]string, 0, len(versions))
+	for k := range versions {
+		keys = append(keys, k)
+	}
```

Comment:
can we add recommended support in this way?
```suggestion
       for k, v := range versions {
               if strings.ToLower(apply) == recommended && v.Status != pbVersion.Status_ recommended {
                       continue
               }
		keys = append(keys, k)
	}
```